src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const separatorToString = separator + '';
+    let resultString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const element = this[i];
+
+      if (element === undefined || element === null) {
+        resultString += separatorToString;
+      } else {
+        resultString += element + separatorToString;
+      }
+    }
+
+    const lengthSeparator = separatorToString.length;
+
+    if (lengthSeparator === 0) {
+      return resultString;
+    }
+
+    const stringWithoutLastElement = resultString.slice(0, -lengthSeparator);
```

Comment:
Could we just check it in the loop ?

Reply:
> Push your code to the github

@solaryasha  To be honest, I have no idea how to put the last check in the loop. I spent about an hour and it didn't work out for me) So I made another solution